### ISSUE TITLE
AWS - Automatically add Tags and TagSets to resources when possible

### ISSUE THREAD

**Issue by klauern** [nccgroup/ScoutSuite#627]
It would be useful for any resource that supports Tags or TagSets to be pulled in automatically.  Right now it is something that is done on a service-by-service basis, as is not consistent across services that are implemented.

There does appear to be a set of resources where the `Tags` are listed as an attribute, as shown from this search across Boto3: https://boto3.amazonaws.com/v1/documentation/api/latest/search.html?q=tags&check_keywords=yes&area=default

**Comment by j4v** [comment]
While I agree, there's currently no way for us to add this in automatically. Resources are being parsed individually, so adding this will/would require updating all resources being parsed.

**Comment by klauern** [comment]
Ah, fair enough.  I will look at possibly adding that in, or at least simplifying how you add it in to all the resources.

**Comment by j4v** [comment]
Similar to https://github.com/nccgroup/ScoutSuite/issues/660, but for AWS

**Comment by j4v** [comment]
If all resources have this in a similar field we could create a generic method to pull & display this field.

**Comment by lowSoA** [comment]
Found some documentation regarding this.
https://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/Welcome.html

At the bottom there is a section with "Services that support the Resouce Groups Tagging API". Since I'm not familiar with this feature, does that list include all the resources that can be tagged?

**Comment by klauern** [comment]
@lowSoA I think that is referring to performing tagging on resources.  This ticket is that the Tags from a resource aren't being inserted into the query data from ScoutSuite, and from what @j4v mentioned, to add those tags in would require adding to each of the facade API's the callout to map or include the Tags.

So really, each resource already has the Tags.  It just isn't making it's way through the pipeline and in the result JSON, which is what I was hoping we could add logic to that globally.  It appears that there's no global hook at the AWS API level to insert a base set of types or fields into all the data.

**Comment by j4v** [comment]
Yeah, looks like we'll need to go through the resources and make sure the tags are included (@lowSoA similar to what you did for Azure tags and RGs).
